packages/cms/pages/settings/profile/profile.tsx

```diff
@@ -38,24 +44,86 @@ const SpacerBottom = styled(EuiSpacer)`
 
 interface ProfileProps {}
 
+type UpdateProfileErrors = Partial<
+  Record<keyof UpdateProfileCredentials, string[]>
+>
+
 export const Profile: FunctionComponent<ProfileProps> = () => {
+  const { user, setUser, updateProfile } = useAuthStore()
+  const [profile, setProfile] = useState<UpdateProfileCredentials>()
+  const [pageHasLoaded, setPageLoaded] = useState(false)
+  const [isUpdatingProfile, setUpdatingProfile] = useState(false)
+  const [
+    updateProfileErrors,
+    setUpdateProfileErrors
+  ] = useState<UpdateProfileErrors>({})
+  const [toasts, setToasts] = useState<any>([])
+
+  let toastId = 0
+
+  useEffect(() => {
+    if (!pageHasLoaded) {
+      const { firstName, lastName, email } = user
+      setProfile({ firstName, lastName, email })
+      setPageLoaded(true)
+    }
+  })
+
+  const onUpdateProfile = async (onSubmitEvent: FormEvent<HTMLFormElement>) => {
+    onSubmitEvent.preventDefault()
+
+    setUpdatingProfile(true)
+    const [response, error] = await updateProfile(profile!)
+
+    if (error) {
+      let errorData = error.response?.data?.errors
+
+      let errors: UpdateProfileErrors = {}
+
+      // get error messages
+      errorData.forEach((error: { message: string; field: string }) => {
+        errors[error.field as keyof UpdateProfileErrors] = [error.message]
+      })
+
+      setUpdateProfileErrors({ ...errors })
+      setUpdatingProfile(false)
+      return
+    }
+
+    setUpdatingProfile(false)
+    setUser({ ...user, ...profile })
+    const toast = {
```

Comment:
For toast messages, we're gonna take a different approach. Now that you started it, we can implement it once and for all. So we want a global approach instead.

1. Create a new store called `toasts.ts`
2. In this store, let's have `toasts` which is an array of toasts. Let's also have a method called `toast`. This toast method should receive `title`, `color` and optional `message` as arguments.
3. At the project root, like the `App` component, mount the `EuiGlobalToastList` component there.
4. Now use the toasts store, get the toasts from the store and pass to this global toast list component.

In the end, in this component, we should be able to fire toasts like this:

```ts

const { toast } = useToastsStore()


toast('Profile updated successfully.', 'Your profile information has been updated.', 'success') // success should be default, so optional.
```

Reply:
okay, this would be nice.

Can work on it in a different branch as a fix